### ISSUE TITLE
Error: "Cannot find module 'nx/src/config/workspaces'"

### ISSUE THREAD

**Issue by GrumpyMeow** [Phillip9587/nx-stylelint#399]
Hi,

When running `npx ng generate nx-stylelint:scss --project=dummy-app` in our workspace, this error is reported unfortunatly:
```
Cannot find module 'nx/src/config/workspaces'
Require stack:
- /home/me/.../node_modules/nx-stylelint/node_modules/@nrwl/devkit/index.js
- /home/me/.../node_modules/nx-stylelint/src/generators/scss/generator.js
- /home/me/.../node_modules/nx/src/shared/workspace.js
- /home/me/.../node_modules/nx/src/cli/init-local.js
- /home/me/..../node_modules/nx/bin/nx.js
```

```
   Node : 16.14.2
   OS   : linux x64
   npm  : 8.5.0
   
   nx : 13.9.5
   @nrwl/angular : 13.9.5
   @nrwl/cypress : 13.9.5
   @nrwl/detox : Not Found
   @nrwl/devkit : 13.9.5
   @nrwl/eslint-plugin-nx : 13.9.5
   @nrwl/express : 13.9.5
   @nrwl/jest : 13.9.5
   @nrwl/js : 13.9.5
   @nrwl/linter : 13.9.5
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : 13.9.5
   @nrwl/nx-cloud : 14.0.3
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 13.9.5
   @nrwl/web : 13.9.5
   @nrwl/workspace : 13.9.5
   typescript : 4.5.5
   rxjs : 7.5.7
   ---------------------------------------
   Community plugins:
         @fortawesome/angular-fontawesome: 0.10.0
         @ngrx/component-store: 13.0.2
         @ngrx/effects: 13.0.2
         @ngrx/entity: 13.0.2
         @ngrx/router-store: 13.0.2
         @ngrx/store: 13.0.2
         @ngrx/schematics: 13.0.2
         @ngrx/store-devtools: 13.0.2
         @storybook/angular: 6.5.12
         ng-mocks: 14.2.0
         nx-stylelint: 14.0.2
```

```
npm ls stylelint
├─┬ nx-stylelint@14.0.2
│ └── stylelint@14.13.0 deduped
├─┬ stylelint-config-prettier@9.0.3
│ └── stylelint@14.13.0 deduped
├─┬ stylelint-config-recommended-scss@7.0.0
│ ├─┬ stylelint-config-recommended@8.0.0
│ │ └── stylelint@14.13.0 deduped
│ └── stylelint@14.13.0 deduped
├─┬ stylelint-config-standard@28.0.0
│ ├─┬ stylelint-config-recommended@9.0.0
│ │ └── stylelint@14.13.0 deduped
│ └── stylelint@14.13.0 deduped
├─┬ stylelint-prettier@2.0.0
│ └── stylelint@14.13.0 deduped
├─┬ stylelint-scss@4.3.0
│ └── stylelint@14.13.0 deduped
└── stylelint@14.13.0
```

I suspect it's a compatibility issue between NX and the plugin.

When i run the stylelint target i also get the following error:
```
node:internal/process/promises:279
            triggerUncaughtException(err, true /* fromPromise */);
            ^
{
  stack: "TypeError: Cannot read properties of undefined (reading 'endsWith')\n" +
    '    at createProjectRootMappings (/home/.../.../node_modules/@nrwl/workspace/src/core/file-map-utils.js:12:38)\n' +
    '    at createProjectFileMap (/home/.../.../node_modules/@nrwl/workspace/src/core/file-map-utils.js:27:33)\n' +
    '    at /home/.../.../node_modules/@nrwl/workspace/src/core/project-graph/daemon/server/project-graph-incremental-recomputation.js:92:85\n' +
    '    at Generator.next (<anonymous>)\n' +
    '    at fulfilled (/home/.../.../node_modules/tslib/tslib.js:114:62)\n' +
    '    at processTicksAndRejections (node:internal/process/task_queues:96:5)',
  message: "Cannot read properties of undefined (reading 'endsWith')\n" +
    '\n' +
    'Because of the error the Nx daemon process has exited. The next Nx command is going to restart the daemon process.\n' +
    'If the error persists, please run "nx reset".'
```

Darn. I'll try the plugin after upgrading to NX14/NX15 i guess.

Seems also to be because of this attribute to be removed from the project.json:
`"root": "apps/dummy-app",`

**Comment by Phillip9587** [comment]
Hi, 

I think there is a version mismatch. Please take a look at the compatibility table: https://github.com/Phillip9587/nx-stylelint#compatibility-with-nx-and-stylelint
